File: medium/1456. Maximum Number of Vowels in a Substring of Given Length/1456. Maximum Number of Vowels in a Substring of Given Length.go
package main

import (
	"slices"
)

func maxVowels(s string, k int) int {
	vowels := []byte{'a', 'e', 'i', 'o', 'u'}
	count := 0
	res := -1

	for i := 0; i < k-1; i++ {
		if slices.ContainsFunc(vowels, func(v byte) bool {
			return s[i] == v
		}) {
			count += 1
		}
	}

	n := len(s)
	left := 0

	for right := k - 1; right < n; right++ {
		if slices.ContainsFunc(vowels, func(v byte) bool {
			return s[right] == v
		}) {
			count += 1
		}

		res = max(res, count)

		if slices.ContainsFunc(vowels, func(v byte) bool {
			return s[left] == v
		}) {
			count -= 1
		}

		left += 1
	}

	return res
}

File: medium/1456. Maximum Number of Vowels in a Substring of Given Length/1456. Maximum Number of Vowels in a Substring of Given Length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        n = len(s)
        vowels = {"a", "e", "i", "o", "u"}
        count = 0
        res = -1

        for i in range(k - 1):
            if s[i] in vowels:
                count += 1

        left = 0
        for right in range(k - 1, n):
            if s[right] in vowels:
                count += 1

            res = max(res, count)

            if s[left] in vowels:
                count -= 1

            left += 1

        return res
